### ISSUE TITLE
Cannot compile with VitisAI

### ISSUE THREAD

**Issue by HemanLin-cl** [amd/RyzenAI-SW#222]
Dear authors, 

I've tried the sample code under "tutorial/getting_started_resnet/bf16/predict.py" 
It can run into NPU and infer successfully by feeding a fp32 resnet model and set "--ep npu" to compile model. 

However, when I use my own fp32 model (transformer-like segmentation), it crashed. 
This is what I set python "predict.py --ep npu" <-- replace onnx_model_path into my own model path 

These are the operators of my model: 
['Conv', 'Constant', 'Div', 'Erf', 'Add', 'Mul', 'BatchNormalization', 'Shape', 'Slice', 'Concat', 'Reshape', 'Transpose', 'MatMul', 'Split', 'Squeeze', 'Softmax', 'Relu', 'Unsqueeze', 'Resize', 'Gather', 'Cast', 'Sigmoid', 'Sub', 'Clip', 'Neg', 'Abs', 'Equal', 'If', 'ConstantOfShape', 'Pad', 'ReduceMin', 'ReduceMax']

The followings are the messages: 
===========================================================================================
execution started on NPU
WARNING: Logging before InitGoogleLogging() is written to STDERR
I20250806 17:33:24.138381 20740 vitisai_compile_model.cpp:1157] Vitis AI EP Load ONNX Model Success
I20250806 17:33:24.138381 20740 vitisai_compile_model.cpp:1158] Graph Input Node Name/Shape (1)
I20250806 17:33:24.138381 20740 vitisai_compile_model.cpp:1162]          image : [1x3x1024x1024]
I20250806 17:33:24.138381 20740 vitisai_compile_model.cpp:1168] Graph Output Node Name/Shape (1)
I20250806 17:33:24.138381 20740 vitisai_compile_model.cpp:1172]          mask : [1x1x1024x1024]

   Creating library C:\Users\xxx\WINDOW~1\RYZENA~1.5\tutorial\GETTIN~1\bf16\MY_CAC~1\MATTIN~4\VAIML_~1\libflexml_usermodel.lib and object C:\Users\xxx\WINDOW~1\RYZENA~1.5\tutorial\GETTIN~1\bf16\MY_CAC~1\MATTIN~4\VAIML_~1\libflexml_usermodel.exp
LINK : warning LNK4217: symbol 'exit' defined in 'libucrt.lib(exit.obj)' is imported by 'model-codegen-wrappers.obj' in function 'flexml_model_model'
Exception Code: 0xC0000005
 #0 0x00007fff444820e1 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0x42620e1)
 #1 0x00007fff4460f01e (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0x43ef01e)
 #2 0x00007fff4460e2d7 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0x43ee2d7)
 #3 0x00007fff4460d681 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0x43ed681)
 #4 0x00007fff4460d475 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0x43ed475)
 #5 0x00007fff420283e5 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0x1e083e5)
 #6 0x00007fff4063986e (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0x41986e)
 #7 0x00007fff4063775f (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0x41775f)
 #8 0x00007fff40295537 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0x75537)
 #9 0x00007fff402954b1 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0x754b1)
#10 0x00007fff406372f4 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0x4172f4)
#11 0x00007fff41f6d685 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0x1d4d685)
#12 0x00007fff41f6dd4f (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0x1d4dd4f)
#13 0x00007fff41f6f5e1 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0x1d4f5e1)
#14 0x00007fff41f6d4e8 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0x1d4d4e8)
#15 0x00007fff41f70865 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0x1d50865)
#16 0x00007fff4027c9e5 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0x5c9e5)
#17 0x00007fff40272202 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0x52202)
#18 0x00007fff402b9387 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0x99387)
#19 0x00007fff403b94c8 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0x1994c8)
#20 0x00007fff4031fb6d (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\flexml\flexml_extras\lib\vaiml.dll+0xffb6d)
#21 0x00007fff6856e6cf (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_vitisai_ep.dll+0x27ee6cf)
#22 0x00007fff6801e3bc (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_vitisai_ep.dll+0x229e3bc)
#23 0x00007fff67f7746c (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_vitisai_ep.dll+0x21f746c)
#24 0x00007fff67f765a0 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_vitisai_ep.dll+0x21f65a0)
#25 0x00007fff67f7b8de (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_vitisai_ep.dll+0x21fb8de)
#26 0x00007fff681e1780 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_vitisai_ep.dll+0x2461780)
#27 0x00007fff681e8026 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_vitisai_ep.dll+0x2468026)
#28 0x00007fff681e8422 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_vitisai_ep.dll+0x2468422)
#29 0x00007fff6822b74f (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_vitisai_ep.dll+0x24ab74f)
#30 0x00007fff6821f6b0 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_vitisai_ep.dll+0x249f6b0)
#31 0x00007fff68220a5c (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_vitisai_ep.dll+0x24a0a5c)
#32 0x00007fff6821ff8d (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_vitisai_ep.dll+0x249ff8d)
#33 0x00007fffec740e12 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_providers_vitisai.dll+0x10e12)
#34 0x00007fffec74ddee (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_providers_vitisai.dll+0x1ddee)
#35 0x00007fff74ff16cd (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_pybind11_state.pyd+0x9b16cd)
#36 0x00007fff74ff5940 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_pybind11_state.pyd+0x9b5940)
#37 0x00007fff74ff54e9 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_pybind11_state.pyd+0x9b54e9)
#38 0x00007fff74ff4fd8 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_pybind11_state.pyd+0x9b4fd8)
#39 0x00007fff74743807 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_pybind11_state.pyd+0x103807)
#40 0x00007fff747347ac (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_pybind11_state.pyd+0xf47ac)
#41 0x00007fff746f5dd2 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_pybind11_state.pyd+0xb5dd2)
#42 0x00007fff746eacbf (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_pybind11_state.pyd+0xaacbf)
#43 0x00007fff746c4106 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_pybind11_state.pyd+0x84106)
#44 0x00007fff74654b44 (C:\Users\xxx\miniconda3\envs\resnet_bf16\lib\site-packages\onnxruntime\capi\onnxruntime_pybind11_state.pyd+0x14b44)
#45 0x00007fff7a876486 cfunction_call \objects\methodobject.c:544:0
#46 0x00007fff7a833aec _PyObject_MakeTpCall \objects\call.c:215:0
#47 0x00007fff7a835da9 method_vectorcall \objects\classobject.c:55:0
#48 0x00007fff7a917cb9 _PyObject_VectorcallTstate \include\cpython\abstract.h:115:0
#49 0x00007fff7a9200e2 call_function \python\ceval.c:5894:0
#50 0x00007fff7a91c94f _PyEval_EvalFrameDefault \python\ceval.c:4191:0
#51 0x00007fff7a91e8cb _PyEval_Vector \python\ceval.c:5080:0
#52 0x00007fff7a833dfe _PyFunction_Vectorcall \objects\call.c:347:0
#53 0x00007fff7a917cb9 _PyObject_VectorcallTstate \include\cpython\abstract.h:115:0
#54 0x00007fff7a9200e2 call_function \python\ceval.c:5894:0
#55 0x00007fff7a91c973 _PyEval_EvalFrameDefault \python\ceval.c:4207:0
#56 0x00007fff7a91e8cb _PyEval_Vector \python\ceval.c:5080:0
#57 0x00007fff7a833dfe _PyFunction_Vectorcall \objects\call.c:347:0
#58 0x00007fff7a833931 _PyObject_FastCallDictTstate \objects\call.c:155:0
#59 0x00007fff7a834072 _PyObject_Call_Prepend \objects\call.c:431:0
#60 0x00007fff7a89e46b slot_tp_init \objects\typeobject.c:7714:0
#61 0x00007fff7a8921bb type_call \objects\typeobject.c:1129:0
#62 0x00007fff7a833aec _PyObject_MakeTpCall \objects\call.c:215:0
#63 0x00007fff7a9200e2 call_function \python\ceval.c:5894:0
#64 0x00007fff7a91c9b3 _PyEval_EvalFrameDefault \python\ceval.c:4240:0
#65 0x00007fff7a91e8cb _PyEval_Vector \python\ceval.c:5080:0
#66 0x00007fff7a833dfe _PyFunction_Vectorcall \objects\call.c:347:0
#67 0x00007fff7a917cb9 _PyObject_VectorcallTstate \include\cpython\abstract.h:115:0
#68 0x00007fff7a9200e2 call_function \python\ceval.c:5894:0
#69 0x00007fff7a91c2fd _PyEval_EvalFrameDefault \python\ceval.c:4222:0
#70 0x00007fff7a91e8cb _PyEval_Vector \python\ceval.c:5080:0
#71 0x00007fff7a919372 PyEval_EvalCode \python\ceval.c:1135:0
#72 0x00007fff7a98f19e run_eval_code_obj \python\pythonrun.c:1290:0
#73 0x00007fff7a98f278 _Py_DECREF \include\object.h:492:0
#74 0x00007fff7a98f278 run_mod \python\pythonrun.c:1311:0
#75 0x00007fff7a98ee28 pyrun_file \python\pythonrun.c:1206:0
#76 0x00007fff7a98d105 _PyRun_SimpleFileObject \python\pythonrun.c:455:0
#77 0x00007fff7a98c2a4 _PyRun_AnyFileObject \python\pythonrun.c:89:0
#78 0x00007fff7a7b7a3c pymain_run_file_obj \modules\main.c:358:0
#79 0x00007fff7a7b844d pymain_run_python \modules\main.c:591:0
#80 0x00007fff7a7b92d3 pymain_main \modules\main.c:1059:0
#81 0x00007fff7a7b9346 Py_Main \modules\main.c:1072:0
#82 0x00007ff7b52d1494 invoke_main d:\agent\_work\2\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:90:0
#83 0x00007ff7b52d1494 __scrt_common_main_seh d:\agent\_work\2\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288:0
#84 0x00007ff80ee5dbe7 (C:\Windows\System32\KERNEL32.DLL+0x2dbe7)
#85 0x00007ff80fc65a6c (C:\Windows\SYSTEM32\ntdll.dll+0x85a6c)
=========================================================================================== 

Are there other modifications I can do to compile my model? 

I've also tried these items: 

(1)  onnxruntime dynamic_quant 
      It can restore the quantized model. And I feed this model into predict.py --ep npu, but it is all run under cpu anyway
   
(2)  Divide the original onnx into smaller one which contains less operators 
      
<img width="502" height="1134" alt="Image" src="https://github.com/user-attachments/assets/80df8a1e-53e9-4ba3-aefc-aa033c989075" />
      
      Use "Mul" marked with red rectangle as output and save the onnx model. 
      And then run python predict.py --ep npu (Replace model with the smaller one) 
      
      Error copying files: copy: The system cannot find the path specified.:   "C:\MATTIN~1\VAIML_~1\0\aiecompiler\Work\ps\ml_txn\ml_txn.bin", "C:\MATTIN~1\VAIML_~1\0\ml_txn.bin"
ERROR: [VAIML-COMPILE 1026] cp error.
      There's no "ps" folder indeed 

      [Vitis AI EP] No. of Operators :   CPU     6 
      *If we set other nodes as outputs, they can all run into VAIML(NPU) indeed 

      I also try to replace "Erf" with "tanh", the above cp errors vanished and crash at other operators which I'm still investigating
      
<img width="564" height="879" alt="Image" src="https://github.com/user-attachments/assets/d69c3708-0a6e-4fb1-8c51-71a10ac1f6c4" />
  
(3) tutorial/quark_quantization
      Use the default config XINT8. It is all run under cpu anyway
      *If I use "A8W8" config, the model will hang for over an hour without crashing and finally run cpu when enable npu  
There're other warnings I've faced during inferencing: 

(1) WARNING:root:Inference failed or unsupported type to quantize for tensor '/Slice_3_output_0'
(2) 
F20250804 19:47:34.242486 24784 KernelParamGenPass.cpp:14168] xir::Op{name = (/decoder4/Wattn/Squeeze_output_0_DequantizeLinear_Output), type = transpose}. This order: (0, 3, 2,
Unhandled exception caught in c10/util/AbortHandler.h 

(3) ONNX Reshape in your model is using Opset 5, which is quite old. Please consider regenerate your model with a newer Opset. Transpose is opset 1. 

Or shall I use onnxruntime-OGA to quantize? 

Is there any suggested default config for my case? (https://quark.docs.amd.com/latest/onnx/user_guide_config_description.html) 

Thanks

**Comment by dwithchenna** [comment]
We're closing this issue due to inactivity. If you're still having this problem, feel free to reopen it and let us know. Thanks!
